client/client.go

```diff
@@ -0,0 +1,166 @@
+package client
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sort"
+
+	rpc "github.com/keybase/go-framed-msgpack-rpc"
+	"github.com/keybase/search/libsearch"
+	sserver1 "github.com/keybase/search/protocol/sserver"
+	"golang.org/x/net/context"
+)
+
+// Client contains all the necessary information for a KBFS Search Client.
+type Client struct {
+	searchCli   sserver1.SearchServerClient   // The client that talks to the RPC Search Server.
+	directory   string                        // The directory where KBFS is mounted.
+	indexer     *libsearch.SecureIndexBuilder // The indexer for the client.
+	pathnameKey [32]byte                      // The key to encrypt and decrypt the pathnames to/from document IDs.
```

Comment:
We should have proper typedefs for the key types, so that later we can make sure for example they are never printed out in the logs (by overriding `String()`).